pkg/reconciler/tenancy/workspace/workspace_reconcile_phase.go

```diff
@@ -106,7 +116,34 @@ func (r *phaseReconciler) reconcile(ctx context.Context, workspace *tenancyv1alp
 			logger.Info("workspace content is deleted")
 			return reconcileStatusContinue, nil
 		}
+
+		// if workspace is ready, we check if it suppose to be ready by checking conditions.
+		if updateTerminalConditionsAndPhase(workspace) {
+			logger.Info("workspace phase changed", "status", workspace.Status)
+			return reconcileStatusStopAndRequeue, nil
+		}
 	}
 
 	return reconcileStatusContinue, nil
 }
+
+// updateTerminalConditionsAndPhase checks if the workspace is ready by checking conditions and sets the phase accordingly.
+// It returns true if the phase was changed, false otherwise.
+func updateTerminalConditionsAndPhase(workspace *tenancyv1alpha1.Workspace) bool {
```

Comment:
this does not change the conditions.

Can we express this functionally without side-effects?
```golang
func terminalConditionPhase(workspace *tenancyv1alpha1.Workspace) (string, bool) {
```
Maybe like that. Easier to reason about.